packages/flutter/lib/src/material/switch.dart

```diff
@@ -644,16 +636,14 @@ class _RenderSwitch extends RenderToggleable {
   _SwitchState state;
 
   @override
-  set value(bool newValue) {
+  set value(bool? newValue) {
     assert(value != null);
     super.value = newValue;
     // The widget is rebuilt and we have pending position animation to play.
     if (_needsPositionAnimation) {
       _needsPositionAnimation = false;
-      position
-        ..curve = null
-        ..reverseCurve = null;
-      if (newValue)
+      position.reverseCurve = null;
```

Comment:
cc @HansMuller we might be able to get rid of this line too, it looks like RenderToggleable always sets reverseCurve when it sets curve. But that's a story for another PR.